### PR TITLE
Adds KYC Wall trigger for Transfer Balance action

### PR DESCRIPTION
**Pull request by marcaaron**
cc @stitesExpensify @nickmurray47 

### Details

- Adds the KYC checks to attempts to transfer a balance
- Fixes up the positioning of the "add payment method" popover

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/187802

### Tests

**Note:** In order to simulate a user instantly passing Onfido + IDOlogy we need to take a few additional steps that are explained [here](https://stackoverflow.com/c/expensify/questions/10607).

1. Start with an account that has a gold wallet + payment method added i.e. able to "Send Money"
2. Send money to an account that does not have a gold wallet
3. As the recipient, navigate to the "Settings > Payments" screen and tap "Transfer Balance"
4. Verify that a popup appears asking you to add a "Bank account" or "Debit card"

5. Select Bank Account + add a test bank account
6. Verify you are navigated to the "Verify identity" flow next

7. Go through Onfido 
8. Enter the required information in Additional details step (follow second answer [here](https://stackoverflow.com/c/expensify/questions/10607/10608#10608))
9. Verify that you are brought to the balance transfer page

- [x] Verify that no errors appear in the JS console

### QA Steps

Internal QA ... maybe Production QA? Not sure if Onfido works on staging.

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Review by stitesExpensify — DISMISSED** [review]

**Review by nickmurray47 — DISMISSED** [review]
not able to test right now because of some Auth/VM issues but code LGTM too!

**Comment by stitesExpensify** [comment]
Will test and merge this morning

**Comment by marcochavezf** [comment]
I got this error in step 3:

Screen Shot 2022-01-19 at 13 04 29

I'm going to test again with a different account.

**Comment by marcochavezf** [comment]
Getting the same issue with another account with Gold wallet. Seems the `userWallet`property is missing here: 

https://github.com/Expensify/App/blob/c857f8b1a5ccf1baf895770f00b2369cec727975/src/pages/settings/Payments/TransferBalancePage.js#L109

Because it's undefined in `PaymentUtils.formatPaymentMethods`:

Screen Shot 2022-01-19 at 13 36 07

**Comment by marcaaron** [comment]
Uh hmm, 

https://github.com/Expensify/App/blob/a2f6349e41eed50344708be04328be1d58be019a/src/pages/settings/Payments/TransferBalancePage.js#L109-L112

We are not passing it here

**Comment by marcaaron** [comment]
Pushed a change to fix this here. Gonna re-test myself (last time I checked I had no issues accessing).

**Comment by marcaaron** [comment]
Hmm, I'm really confused why this is not working all of a sudden... the data is there



I have a feeling this is related to the IndexedDB change...

**Comment by marcaaron** [comment]
Ok it took me _way_ too long to figure out that we need to pass a paypal me name to the `formatPaymentMethods()` method 😂 

The `SvgChase()` error + propTypes were unrelated, but also fixing them here to avoid future confusion.

**Comment by marcaaron** [comment]
Updated.

File: src/pages/settings/Payments/TransferBalancePage.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const propTypes = {
     userWallet: userWalletPropTypes.userWallet,
 
     /** Array of bank account objects */
```
Comment (marcochavezf) [inline src/pages/settings/Payments/TransferBalancePage.js:32]: NAB: Ah I think we should update that this is no longer an array but an object, no?

File: src/pages/settings/Payments/TransferBalancePage.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const propTypes = {
     })),
 
     /** Array of card objects */
```
Comment (marcochavezf) [inline src/pages/settings/Payments/TransferBalancePage.js:47]: NAB: Same here? Which can be an Object (or Dictionary) of card objects?

**Comment by marcaaron** [comment]
Updated.

**Comment by marcochavezf** [comment]
The Transfer Balance page is working fine now :D but for some reason I'm stuck in step 5; the button `Instant (Debit Card)` is selected but the `Transfer` button is disabled also I can't select the other option `1-3 Business Days (Bank Account)`

https://user-images.githubusercontent.com/6829422/150391870-f3704cae-e640-4c04-afbc-d14e1c1d9266.mov

**Comment by marcaaron** [comment]
>for some reason I'm stuck in step 5; the button Instant (Debit Card) is selected but the Transfer button is disabled also I can't select the other option 1-3 Business Days (Bank Account)

Which step 5? The QA for this PR is to just make sure you can get to the "Transfer Balance" page and looks like you were able to do it :)

**Comment by marcaaron** [comment]
I'll add some pics to show what I'm looking for here.

**Comment by marcaaron** [comment]
Just tested the flow again and added screenshots. @marcochavezf let me know if everything looks good to you!

**Comment by marcochavezf** [comment]
Oh I see, thanks! I was using an account that was skipping the "Verify identity" step

**Review by marcochavezf — APPROVED** [review]

**Review by stitesExpensify — APPROVED** [review]
Nice!

**Comment by mvtglobally** [comment]
@marcaaron @marcochavezf @nickmurray47 @stitesExpensify This PR is ready to QA internally

**Comment by marcaaron** [comment]
Thanks, there is an issue where we can't access the production API so we will need to test this on production. This feature is still behind a beta so we can check this off I think.

https://github.com/Expensify/App/issues/7371

**Pull request merged**